lib/puppet/util/servicenow.rb

```diff
@@ -392,6 +392,31 @@ def catalog_compilation_failure?(resource_statuses, transaction_completed)
     resource_statuses.empty? && !transaction_completed
   end
   module_function :catalog_compilation_failure?
+
+  def find_environment_match(environment, allow, block)
+    return true if allow == 'none' || block == 'all'
+    return false if block == 'none' || allow == 'all'
+    return true if File.fnmatch(block, environment)
+    return false if File.fnmatch(allow, environment)
+  rescue e
+    raise 'Error with environment filter. Please look over allow_list and block_list configurations'
```

Comment:
I would return the original message with e.message along with your helpful message.